pkg/api/v20210131preview/vm.go

```diff
@@ -0,0 +1,55 @@
+package v20210131preview
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+func VMSizeIsValid(vmSize VMSize, requiredD2sV3Workers, isMaster bool) bool {
```

Comment:
Wondering if makes sense to get rid of `pkg/api/validate/vm.go` and make a copy for every api version for consistency.

Reply:
@bennerv and I had discussions about this because it will stay with the api version (discussion to a wider audience still needs to take place)

Reply:
This will also impact the machine_checker in the checker controller for the operator.  Currently, it uses the `pkg/validate` `VMSizeIsValid` function which will fail for compute isolation because the vm types are not in the api it references.

Reply:
We can use the `v20210131preview` `VMSizeIsValid` function in the controller as it still has the same vm types with the additional vm types for the compute isolation. Since we are not coupling validation to the api and the preview api changes, this is necessary to cover the preview api changes that allow new vm types.